Tasks/DownloadBuildArtifactsV0/main.ts

```diff
@@ -332,8 +333,32 @@ async function main(): Promise<void> {
                 }
             });
 
-            Promise.all(downloadPromises).then(() => {
+            Promise.all(downloadPromises).then((tickets: models.ArtifactDownloadTicket[][]) => {
                 console.log(tl.loc('ArtifactsSuccessfullyDownloaded', downloadPath));
+
+                const shouldExtractTars: boolean = tl.getBoolInput('extractTars');
+                if (shouldExtractTars) {
+                    const flatTickets: models.ArtifactDownloadTicket[] = [].concat(...tickets);
+                    const tarArchivesPaths = [] as string[];
+
+                    flatTickets.forEach((ticket: models.ArtifactDownloadTicket) => {
+                        if (
+                            ticket.artifactItem.itemType === models.ItemType.File
+                            && ticket.artifactItem.path.endsWith('.tar')
```

Comment:
Do we need null checks here? Could you please double-check that there properties are always not null?

Reply:
Yeah, double-checked this: `ArtifactDownloadTicket` type should never have null properties, and the tickets themselves can't be null either